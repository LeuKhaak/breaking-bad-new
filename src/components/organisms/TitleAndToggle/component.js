import React from "react";
import * as styles from "./styles.module.scss";
import Btn from "src/components/atoms/Btn/index";
import lines from "../../../assets/icons/lines.svg";
import tiles from "../../../assets/icons/tiles.svg";
import lines2 from "../../../assets/icons/lines2.svg";
import tiles2 from "../../../assets/icons/tiles2.svg";

function TitleAndToggle({ tile, changeStyles }) {
  return (
    <div className={styles.container}>
      <h3 className={styles.title}>Catalog</h3>
      <div className={styles.toggleWrapper}>
        <Btn
          btnStyle="toggle"
          icon={tile ? lines : lines2}
          onClick={changeStyles}
          arg={false}
        />
        <Btn
          btnStyle="toggle"
          icon={tile ? tiles : tiles2}
          onClick={changeStyles}
          arg={true}
        />
      </div>
    </div>
  );
}

export default TitleAndToggle;
