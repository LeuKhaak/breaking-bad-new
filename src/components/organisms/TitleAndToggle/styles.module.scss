@import "../../../assets/styles/styles.scss";
.container {
  @include content;
  max-width: 74.5rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 5rem 0 2rem 0;
  .title {
    font-weight: 500;
    font-size: 32px;
  }
  .toggleWrapper {
    width: 4rem;
    display: flex;
    justify-content: space-between;
    align-items: center;
  }
}
