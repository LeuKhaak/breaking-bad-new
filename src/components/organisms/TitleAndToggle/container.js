import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { changeStyle } from "src/store/actions/person";
import TitleAndToggle from "./component";

function TitleAndToggleContainer() {
  const dispatch = useDispatch();
  const tile = useSelector((state) => state.cards.tiles);
  const changeStyles = (arg) => {
    dispatch(changeStyle(arg));
  };
  return <TitleAndToggle tile={tile} changeStyles={changeStyles} />;
}

export const container = TitleAndToggleContainer;
