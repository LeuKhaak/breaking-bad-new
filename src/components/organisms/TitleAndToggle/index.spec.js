import React from "react";
import { shallow } from "enzyme";
import TitleAndToggle from "./component";

let component = null;
const setComponent = (props) => shallow(<TitleAndToggle {...props} />);
beforeEach(() => {
  component = setComponent();
});

describe("TitleAndToggle testing:", () => {
  test("is TitleAndToggle rendered with active lines", () => {
    component = setComponent({
      tile: false,
    });
    expect(component).toMatchSnapshot();
  });
  test("is TitleAndToggle rendered with active tiles", () => {
    component = setComponent({
      tile: true,
    });
    expect(component).toMatchSnapshot();
  });
});
