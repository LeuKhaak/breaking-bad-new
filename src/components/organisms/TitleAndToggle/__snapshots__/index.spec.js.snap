// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleAndToggle testing: is TitleAndToggle rendered with active lines 1`] = `
<div>
  <h3>
    Catalog
  </h3>
  <div>
    <Btn
      arg={false}
      btnStyle="toggle"
      icon="test-file-stub"
    />
    <Btn
      arg={true}
      btnStyle="toggle"
      icon="test-file-stub"
    />
  </div>
</div>
`;

exports[`TitleAndToggle testing: is TitleAndToggle rendered with active tiles 1`] = `
<div>
  <h3>
    Catalog
  </h3>
  <div>
    <Btn
      arg={false}
      btnStyle="toggle"
      icon="test-file-stub"
    />
    <Btn
      arg={true}
      btnStyle="toggle"
      icon="test-file-stub"
    />
  </div>
</div>
`;
